Fix CommandMessage error

dnpwwo opened the pull request:
Split Python Objects into a separate file

dnpwwo committed:
Final function call updated

dnpwwo committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Python-Plugins

dnpwwo committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Python-Plugins

dnpwwo committed:
Continued splitting code into logical files

Fixed CommandMessage Bug as well

dnpwwo committed:
Unix file system fix for notfications

oohlaf commented on hardware/plugins/DelayedLink.h:
```diff
@@ -253,7 +253,7 @@ namespace Plugins {
 						{
 							std::string filename = ent->d_name;
-							if (dirent_is_directory(szLibrary, ent) && (filename.length() > 2))
+							if ((ent->d_type == DT_DIR) && (filename.length() > 2))
```
Please keep dirent_is_directory, it supports symlinks also, not only directories

oohlaf replied:
Also, better to compare with "." and ".." because the directory can also be a single or two letter directory other than the dots.

filename.compare(".") != 0 && filename.compare("..") != 0

oohlaf reviewed (changes requested):
Please keep symlink support
Would be better to check against dots explicitly not on length

dnpwwo committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Python-Plugins

dnpwwo commented:
I've changed the filename check and it will be in a later pull request.

Sym link support has not been removed.  This code only runs on non-Win32 platforms and so only DT_DIR is checked in dirent_is_directory.  I had real problems compiling the delayed link code on Linux due to dirent_is_directory which is why I took it out.

oohlaf commented:
Huh? I don't understand.

You get symlink support for free as I added it some time ago to dirent_is_directory and it is specifically for non-Win32 platforms as the Linux platform has symlinks, not Win32.

I actually tried your Python-Plugin branch and added back the dirent_is_directory and it compiles fine under Linux. I'll create a pull request against your branch that you can merge easily and try for yourself. I'll leave the filename check up to you....

oohlaf commented:
https://github.com/domoticz/domoticz/pull/1254 which you can pull into your own branch and incorporate with your other changes

dnpwwo committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into Python-Plugins

dnpwwo committed:
Add CustomImage support

Add Images diction that holds CustomImages related to the plugin
Add Image class to hold CustomImage details
